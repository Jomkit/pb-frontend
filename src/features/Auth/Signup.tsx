import Frame from '../../components/Frame'
import { IUser } from '../../types'
import SignupForm from './SignupForm'

const Signup = ({signup}: {signup: (values: IUser) => void}) => {
  return (
    <Frame bgColor='bg-green-400' title='Signup'>
        <SignupForm signup={signup} />
    </Frame>
  )
}

export default Signup