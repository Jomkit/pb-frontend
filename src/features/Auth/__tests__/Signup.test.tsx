import { render, waitFor } from "@testing-library/react";
import Signup from "../Signup";
import { ReactNode } from "react";
import userEvent from "@testing-library/user-event";

describe("signup", function() {
    function setup(tsx: ReactNode) {
        return {
            user: userEvent.setup(),
            ...render(tsx)
        }
    }

    const mockSignup = vi.fn();

    it('should render without crashing', () => { 
        render(<Signup signup={mockSignup} />);
    })

    it('should match snapshot', () => { 
        const { asFragment } = render(
            <Signup signup={mockSignup} />
        )

        expect(asFragment()).toMatchSnapshot();
    })

    it('should call signup', async () => { 
        const { user, getByRole, getByLabelText } = setup(
            <Signup signup={mockSignup} />
        )

        const usernameField = getByLabelText("username");
        const passwordField = getByLabelText("password");
        const firstNameField = getByLabelText("firstName");
        const lastNameField = getByLabelText("lastName");
        const emailField = getByLabelText("email");
        await user.type(usernameField, "testUsername");
        await user.type(passwordField, "testPassword");
        await user.type(firstNameField, "testFirstName");
        await user.type(lastNameField, "testLastName");
        await user.type(emailField, "testEmail@email.com");

        const submitBtn = getByRole('button', {name: /Submit/i});
        user.click(submitBtn);
        await waitFor(() => {
            expect(mockSignup).toHaveBeenCalledWith({
                username: "testUsername", 
                password: "testPassword",
                firstName: "testFirstName", 
                lastName: "testLastName", 
                email: "testEmail@email.com"
            });
            expect(mockSignup).toHaveBeenCalled();
        })
    })

    it("Should not call signup if email is wrong format", async () => {
        const { user, getByRole, findByLabelText } = setup(
            <Signup signup={mockSignup} />
        )

        const usernameField = await findByLabelText("username");
        const passwordField = await findByLabelText("password");
        const firstNameField = await findByLabelText("firstName"); 
        const lastNameField = await findByLabelText("lastName");
        const emailField = await findByLabelText("email");
        await user.type(usernameField, "testUsername");
        await user.type(passwordField, "testPassword");
        await user.type(firstNameField, "testFirstName");
        await user.type(lastNameField, "testLastName");
        await user.type(emailField, "banana[Tab]");

        const submitBtn = getByRole("button", {name: /submit/i});
        try{
            user.click(submitBtn);
        } catch(err: any){
            expect(err.message).toBe("Invalid email format");
        }

    })
})