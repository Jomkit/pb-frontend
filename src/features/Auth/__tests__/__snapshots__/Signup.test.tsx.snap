// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`signup > should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="text-center items-start min-h-screen w-screen p-4 bg-green-400"
  >
    <h1>
      Signup
    </h1>
    <div
      class="p-10 justify-center"
    >
      <form
        action="#"
      >
        <div
          class="rounded-lg border border-gray-200 bg-white shadow-md dark:border-gray-700 dark:bg-gray-800 flex flex-col text-start w-1/2 m-auto"
          data-testid="flowbite-card"
        >
          <div
            class="flex h-full flex-col justify-center gap-4 p-6"
          >
            <label
              class="block text-gray-700 text-sm font-bold mb-2"
              for="username"
            >
              username
            </label>
            <input
              class="form-field1 focus:outline-none focus:shadow-outline"
              id="username"
              name="username"
              type="text"
              value=""
            />
            <label
              class="block text-gray-700 text-sm font-bold mb-2"
              for="password"
            >
              password
            </label>
            <input
              class="form-field1 focus:outline-none focus:shadow-outline"
              id="password"
              name="password"
              type="password"
              value=""
            />
            <label
              class="block text-gray-700 text-sm font-bold mb-2"
              for="firstName"
            >
              firstName
            </label>
            <input
              class="form-field1 focus:outline-none focus:shadow-outline"
              id="firstName"
              name="firstName"
              type="text"
              value=""
            />
            <label
              class="block text-gray-700 text-sm font-bold mb-2"
              for="lastName"
            >
              lastName
            </label>
            <input
              class="form-field1 focus:outline-none focus:shadow-outline"
              id="lastName"
              name="lastName"
              type="text"
              value=""
            />
            <label
              class="block text-gray-700 text-sm font-bold mb-2"
              for="email"
            >
              email
            </label>
            <input
              class="form-field1 focus:outline-none focus:shadow-outline"
              id="email"
              name="email"
              type="text"
              value=""
            />
          </div>
        </div>
        <button
          class="btn bg-blue-700 text-white"
          type="submit"
        >
          Submit
        </button>
      </form>
    </div>
  </section>
</DocumentFragment>
`;
